osu.Game/Screens/Edit/Editor.cs

```diff
@@ -489,7 +498,7 @@ public override bool OnExiting(IScreen next)
 
                 if (isNewBeatmap || HasUnsavedChanges)
                 {
-                    dialogOverlay?.Push(new PromptForSaveDialog(confirmExit, confirmExitWithSave));
+                    dialogOverlay?.Push(new PromptForSaveDialog(confirmExit, confirmExitWithSave, cancelPendingDifficultySwitch));
```

Comment:
Maybe `cancelExit()` instead? It'd be more in line with the other methods and perhaps contain more code in the future.